tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -145,8 +200,75 @@ def test_positive_validate_katello_certs_check_output(self, cert_setup, cert_dat
             )
         self.validate_output(result, cert_data)
 
+    @pytest.mark.tier1
+    def test_katello_certs_check_output_wildcard_inputs(self, generate_certs, cert_data):
+        """Validate that katello-certs-check generates correct output with wildcard certs.
+
+        :id: 7f9da806-5b23-11eb-b7ea-d46d6dd3b5b2
+
+        :steps:
+
+            1. Get valid wildcard certs from generate_certs
+            2. Run katello-certs-check with the required valid arguments
+               katello-certs-check -c CERT_FILE -k KEY_FILE -r REQ_FILE
+               -b CA_BUNDLE_FILE
+            3. Assert the output has correct commands with options
+
+        :expectedresults: katello-certs-check should generate correct commands
+         with options.
+        """
+        with get_connection() as connection:
+            result = connection.run(
+                'katello-certs-check -c {} -k {} -b {}'.format(
+                    "certs/wildcard.crt",
+                    "certs/wildcard.key",
+                    "certs/ca.crt",
+                ),
+                output_format='plain',
+            )
+        self.validate_output(result, cert_data)
+
+    @pytest.mark.parametrize("error, cert_file, key_file, ca_file", invalid_inputs)
+    @pytest.mark.tier1
+    def test_katello_certs_check_output_invalid_input(
+        self, generate_certs, error, cert_file, key_file, ca_file
```

Comment:
Why doesn't this test function include use of `certs_cleanup`?

Reply:
It is handled in the shell file hence no need.

Reply:
I think its a better pattern to clean up all the certs, regardless of invalid or valid.

How is the script cleanup handled in the shell file?  I don't see any commands there that would remove certificates. The shell file is only loaded and run with generate_certs during test setup.  What happens when a statement within the test function fails?